spark-cassandra-connector/src/test/scala/com/datastax/spark/connector/util/ReflectionUtilSpec.scala

```diff
@@ -11,6 +14,25 @@ class ReflectionUtilSpec extends FlatSpec with Matchers {
     factory should be(DefaultConnectionFactory)
   }
 
+  it should "be able to find DefaultConnectionFactory in a multi-threaded context" in {
+    val conductor = new Conductor
+    import conductor._
+
+    thread("ThreadA") {
```

Comment:
For whatever reason this doesn't fail for me if the ^ above test runs first which is why i had it using the custom class